contexts/MembershipContext/src/Domain/Model/Application.php

```diff
@@ -166,4 +166,16 @@ public function isDeleted(): bool {
 		return $this->isDeleted;
 	}
 
+	public function setFirstPaymentDate( string $firstPaymentDate ) {
+		/** @var PayPalPayment $payPalPayment */
+		$payPalPayment = $this->getPayment()->getPaymentMethod();
+		$payPalData = $payPalPayment->getPayPalData();
+		if ( $payPalData === null ) {
+			$payPalData = new PayPalData();
+		}
+
+		$payPalData->setFirstPaymentDate( $firstPaymentDate );
+		$this->addPayPalData( $payPalData );
```

Comment:
If I remember correctly, we wanted to keep the data blob field clear of unset fields. `PayPalData` is actually being set during the payment notification sent by PayPal. Apart from that I don't see any reason not to use `PayPalPayment` for it.